### ISSUE TITLE
How important is a consistent loss to the generalizability?

### ISSUE THREAD

**Issue by meder-yembo** [intel-isl/MiDaS#51]
Do you think the same multi-objective training with different loss functions would yield the same type of generalizable performance. In MiDaS, all training was directly supervised with the same loss function, regardless of dataset. Have you considered or evaluated what the impact of blending different losses would be? For example, take a SSIM loss like Monodepth for training on stereo image pairs or video frames while also training on the datasets you analyzed in this work. From the [original paper](https://arxiv.org/abs/1810.04650), it seems like it should work (they evaluate in a truly multi-task setting), but I am curious if the cross-dataset generalizability would hold up as well.

**Comment by ranftlr** [comment]
I don't see a reason why different losses would be problematic. The original paper doesn't make assumptions about the losses, and as you said, demonstrated their results in the true multi-task setting which matches what you are asking for. We didn't evaluate this empirically for Monodepth, however.

**Comment by meder-yembo** [comment]
Got it. Yeah, I was curious if that was an avenue you'd explored at any point. Thanks!
